test/Microsoft.AspNetCore.Http.Abstractions.Tests/UsePathBaseExtensionsTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder.Internal;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Builder.Extensions
+{
+    public class UsePathBaseExtensionsTests
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("/")]
+        public void EmptyOrNullPathBase_DoNotAddMiddleware(string pathBase)
+        {
+            // Arrange
+            var useCalled = false;
+            var builder = new ApplicationBuilderWrapper(CreateBuilder(), () => useCalled = true)
+                .UsePathBase(pathBase);
+
+            // Act
+            builder.Build();
+
+            // Assert
+            Assert.False(useCalled);
+        }
+
+        private class ApplicationBuilderWrapper : IApplicationBuilder
```

Comment:
It seems to be a general test component. Can you move it to its own file and make it a internal class so other tests can utilize it?